openshift.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# include the common library
+source "$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)/lib/common.sh"
+
+TEMPDIR=$(mktemp -d)
+
+# install the OpenShift cli & virtctl binary
+sudo dnf -y install wget jq
+# https://docs.openshift.com/container-platform/4.13/cli_reference/openshift_cli/getting-started-cli.html
+wget --no-check-certificate https://downloads-openshift-console.apps.ocp-virt.prod.psi.redhat.com/amd64/linux/oc.tar --directory-prefix "$TEMPDIR"
+# https://docs.openshift.com/container-platform/4.13/virt/virt-using-the-cli-tools.html
+wget --no-check-certificate https://hyperconverged-cluster-cli-download-openshift-cnv.apps.ocp-virt.prod.psi.redhat.com/amd64/linux/virtctl.tar.gz --directory-prefix "$TEMPDIR"
+pushd "$TEMPDIR"
+tar -xvf oc.tar
+tar -xzvf virtctl.tar.gz
+popd
+OC_CLI="$TEMPDIR/oc"
+VIRTCTL="$TEMPDIR/virtctl"
+chmod a+x "$OC_CLI"
+chmod a+x "$VIRTCTL"
+
+
+# Authenticate via the gitlab-ci service account
+# oc describe secret gitab-ci-token-g7sw2
+$OC_CLI login --token="$OPENSHIFT_TOKEN" --server=https://api.ocp-virt.prod.psi.redhat.com:6443 --insecure-skip-tls-verify=true
+$OC_CLI whoami
+
+OPENSHIFT_PROJECT="image-builder"
+$OC_CLI project $OPENSHIFT_PROJECT
+
+# iterate over VMs - note: remove the .spec.template field b/c cloud-init data contains
+# unescaped newlines which break echo
+ALL_VMS=$($OC_CLI get vm --output json | jq 'del(.items[].spec.template)' | jq 'del(.items[].status.volumeSnapshotStatuses)' | jq -c '.items[]')
+for INSTANCE in ${ALL_VMS}; do
+    REMOVE=1
+    VM_NAME=$(echo "${INSTANCE}" | jq -r '.metadata.name')
+    CREATION_TIME=$(echo "${INSTANCE}" | jq -r '.metadata.creationTimestamp')
+
+    if [[ $(date -d "${CREATION_TIME}" +%s) -gt "${DELETE_TIME}" ]]; then
+        REMOVE=0
+        echo "The VM '${VM_NAME}' was launched less than ${HOURS_BACK} hours ago"
+    fi
+
+    # TODO: check for tag persist=true
```

Comment:
Got a ticket for this or do you plan to add it in this PR?

Reply:
I was not planning to add the tag functionality as part of this PR. Documented as https://issues.redhat.com/browse/COMPOSER-2195